src/main/java/com/google/api/gax/core/PageAccessor.java

```diff
@@ -0,0 +1,21 @@
+package com.google.api.gax.core;
+
+/**
+ * Accessor for paged results from a list API method"
+ */
+public interface PageAccessor<T> extends Iterable<T> {
```

Comment:
Also, I think you should make it clear that iteration could throw runtime exception (if possible of type ApiException) if fetching the next page failed.